h2quic/server.go

```diff
@@ -72,31 +70,41 @@ func (s *Server) serveImpl(tlsConfig *tls.Config, conn *net.UDPConn) error {
 	if s.Server == nil {
 		return errors.New("use of h2quic.Server without http.Server")
 	}
-	s.serverMutex.Lock()
-	if s.server != nil {
-		s.serverMutex.Unlock()
+	s.listenerMutex.Lock()
```

Comment:
Unfortunately, this doesn't work here. `quic.Listener.Serve`blocks indefinitely, which means that the `defer` will never be executed, and the mutex would never be released.

I created a minimal working example in the Go Playground (https://play.golang.org/p/uRP3khNBBF).